frontend/src/shared/helpers.tsx

```diff
@@ -235,6 +236,39 @@ export const PatientsListColumns: ColDef[] = [
   },
 ];
 
+function LoadingIcon() {
+  return (
+    <div className="lds-ring">
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+    </div>
+  );
+}
+
+const statusCellRendererProps: ColDef = {
+  cellRenderer: (params: ICellRendererParams) => {
+    if (params.data?.revisable) {
+      return params.data?.hasStatusStatuses[0]?.validationStatus ? (
+        <CheckIcon />
+      ) : (
+        <WarningIcon />
+      );
+    } else {
+      return <LoadingIcon />;
+    }
+  },
+  cellRendererParams: {
+    colDef: {
+      tooltipComponent: StatusTooltip,
+      tooltipValueGetter: (params: ITooltipParams) =>
+        params.data.hasStatusStatuses[0]?.validationReport ??
+        params.data.hasStatusStatuses,
+    },
+  },
+};
```

Comment:
This change does two things:
1. Put these column definitions into a named variable `statusCellRendererProps` so that we can pass it into the Status column definition of both the Cohort Sample tables and regular Sample tables
2. Refactor some codes so that we can more easily read and understand when certain icon is used (i.e. check icon, warning icon, and loading icon)